core/src/main/java/io/github/tcdl/msb/monitor/aggregator/DefaultChannelMonitorAggregator.java

```diff
@@ -71,31 +74,52 @@ void onHeartbeatResponses(List<Message> heartbeatResponses) {
         for (Message msg : heartbeatResponses) {
             aggregateInfo(aggregatorStats, msg);
```

Comment:
And yeah, I can explain you why onAnnonce adds the response while onHeartbeatResponses overwrites it. The thing is that this aggregator can work in 2 modes (this logic is taken from Node msb): 
- listen only that only accepts announcements but doesn't send periodic heartbeats
- above + sending periodic heartbeats. In this case we want to overwrite stats with most recent information to detect that some microservices went down.